ocaml/xapi/xapi_vtpm.ml

```diff
@@ -27,19 +27,6 @@ let assert_not_restricted ~__context =
   | _ ->
       raise Api_errors.(Server_error (feature_restricted, [feature]))
 
-(** The state in the xapi backend is only up-to-date when the VMs are halted *)
-let assert_no_fencing ~__context ~persistence_backend =
-  let pool = Helpers.get_pool ~__context in
-  let ha_enabled = Db.Pool.get_ha_enabled ~__context ~self:pool in
-  let clustering_enabled = Db.Cluster.get_all ~__context <> [] in
-  let may_fence = ha_enabled || clustering_enabled in
```

Comment:
This is the one we discussed earlier today.
Perhaps a solution would be to have the equivalent of a mini-redolog active whenever clustering is enabled, using Dom0's filesystem for the redo log. That would not require any action from the user to chose a shared storage for the redo log (although of course if they do chose one then we can use that), and can be done as part of enabling clustering internally in XAPI to also enable this internal redo log mode.

But I wouldn't want to tie together making that change with vTPM, using clustering without enabling any form of redo log or HA is already known to be a bad idea.